internal/pkg/manifest/lb_web_svc_test.go

```diff
@@ -727,7 +736,60 @@ func TestLoadBalancedWebService_ApplyEnv(t *testing.T) {
 							Placement: PlacementArgOrString{
 								PlacementString: placementStringP(PublicSubnetPlacement),
 							},
-							SecurityGroups: []string{"sg-456", "sg-789"},
+							SecurityGroups: SecurityGroupsIDsOrConfig{
+								IDs: []string{"sg-456", "sg-789"},
+							},
+						},
+					},
+				},
+			},
+		},
+		"with network config overridden by security group config": {
+			in: &LoadBalancedWebService{
+				LoadBalancedWebServiceConfig: LoadBalancedWebServiceConfig{
+					Network: NetworkConfig{
+						VPC: vpcConfig{
+							Placement: PlacementArgOrString{
+								PlacementString: placementStringP(PublicSubnetPlacement),
+							},
+							SecurityGroups: SecurityGroupsIDsOrConfig{
+								AdvancedConfig: SecurityGroupsConfig{
+									SecurityGroups: []string{"sg-535", "sg-789"},
+								},
+							},
+						},
+					},
+				},
+				Environments: map[string]*LoadBalancedWebServiceConfig{
+					"prod-iad": {
+						Network: NetworkConfig{
+							VPC: vpcConfig{
+								SecurityGroups: SecurityGroupsIDsOrConfig{
+									AdvancedConfig: SecurityGroupsConfig{
+										SecurityGroups: []string{"sg-456", "sg-700"},
+										DenyDefault:    &trueValue,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			envToApply: "prod-iad",
+
+			wanted: &LoadBalancedWebService{
+				LoadBalancedWebServiceConfig: LoadBalancedWebServiceConfig{
+					Network: NetworkConfig{
+						VPC: vpcConfig{
+							Placement: PlacementArgOrString{
+								PlacementString: placementStringP(PublicSubnetPlacement),
+							},
+							SecurityGroups: SecurityGroupsIDsOrConfig{
+								AdvancedConfig: SecurityGroupsConfig{
+									SecurityGroups: []string{"sg-456", "sg-700"},
+									DenyDefault:    &trueValue,
```

Comment:
nit: related to using `aws.Bool` to take the pointer
```suggestion
									DenyDefault:    aws.Bool(true),
```